dnadoh/recombine.py

```diff
@@ -0,0 +1,50 @@
+"""Recombination of the genomes of two people
+to produce a third person. Other features are
+averaged with some noise.
+"""
+
+from typing import List
+
+def recombine(
+    person1: Person,
+    person2: Person,
+    reference_genome: str,
+    person_gen: PersonGenerator,
+    pid: str,
+    recombination_prob: List[float],
+    recombination_loci: List[int]
+    ) -> Person:
+    """Recombine two genomes"""
+    # Error checking
+    assert len(recombination_loci) == len(recombination_prob)
+    assert len(person1.genome) == len(person2.genome)
+    assert all(recombination_loci >= 0 & recombination_loci < len(person1.genome))
+    assert all(recombination_prob <= 1 & recombination_prob >= 0)
+
+    # Initializing
+    genome1 = person1.genome
+    genome2 = person2.genome
+
+    if 0 not in recombination_loci:
+        recombination_loci = [0] + recombination_loci
+        recombination_prob = [0.5] + recombination_prob
+    if len(genome1) not in recombination_loci:
```

Comment:
why this?

Reply:
Adding the start [0] and endpoints len(genome1) of the genome in order to create intervals. The first recombination probability is 0.5 because I assume I can select either genome as the template with equal probability unless indicated otherwise by the user.